### ISSUE TITLE
Unable to capture input value from cancelled action

### ISSUE THREAD

**Issue by WesCook** [sweetalert2/sweetalert2#2532]
This is somewhere between a bug report, question, and feature request.  I'll use the bug report template but hope that I'm missing something.

## Current behavior

When dismissing a modal with the escape key or clicking outside the window, the input value returns undefined.  The property `returnInputValueOnDeny` does not affect this behaviour (it only affects "decline" actions).

## Expected behavior

I expect there should still be a way to capture the input regardless of how the modal was dismissed.  Either `returnInputValueOnDeny` should affect all actions, or an alternate property should allow this.

## Example

```js
Swal.fire({
	input: "text",
	showCloseButton: true,
	showConfirmButton: true,
	showDenyButton: true,
	showCancelButton: true,
	returnInputValueOnDeny: true
}).then(result => {
	console.log(result.value);
});
```

or

```js
const { value: text } = await Swal.fire({
	input: "text",
	showCloseButton: true,
	showConfirmButton: true,
	showDenyButton: true,
	showCancelButton: true,
	returnInputValueOnDeny: true
});
console.log(text);
```

![alert](https://user-images.githubusercontent.com/792101/196902302-e190cbc2-bf47-4b8e-b27b-4a28d6f5a70b.png)

OK: returns value
NO: returns value if `returnInputValueOnDeny` is set, false otherwise
CANCEL: returns undefined
CLOSE_BUTTON: returns underfined
ESCAPE_KEY: returns underfined
CLICK_OUTSIDE: returns undefined

## Use Case

I'm creating a minimal notepad widget for another application.  It loads the last text when opening, and saves it when closing.  The problem is I can't get it to pass data through the intended close actions (the X button, hitting escape, or clicking outside).  As for why I'm using SweetAlert for this, it handles all the accessibility, prevents background scrolling, and the styling matches the existing modals.

![input](https://user-images.githubusercontent.com/792101/196900867-be17db0f-b2da-4f69-a070-de39f2251a25.png)

Thank you.

**Comment by limonte** [comment]
```js
Swal.fire({
  input: 'text',
}).then(function (result) {
  if (result.isDismissed) {
    // store the input value
    console.log(Swal.getInput().value)
  }
})
```

**Comment by WesCook** [comment]
Aha, I didn't know the input itself was still available after the chained function.  That'll sort me out, thank you.

**Comment by WesCook** [comment]
I'm afraid I ran into another issue where the "Close" workflow doesn't quite work.

Before closing my modal with the "notepad", I'd like to check that the server has received and verified the response.  I can put that in a try/catch to watch for an error, and use that to prevent the modal from closing.  This would be easy with the `preConfirm` or `preDeny` methods, but I don't think it works with the closing method.

Once again I must ask if I'm missing something here, or if this isn't actually possible.  I did notice #2021 discussed this use case, but it was over two years ago so things may have changed.

Thank you again.
